packages/x-date-pickers/src/PickersTextField/PickersOutlinedInput/PickersOutlinedInput.tsx

```diff
@@ -57,19 +55,37 @@ const PickersOutlinedInputRoot = styled(PickersInputBaseRoot, {
     [`&.${pickersOutlinedInputClasses.error} .${pickersOutlinedInputClasses.notchedOutline}`]: {
       borderColor: (theme.vars || theme).palette.error.main,
     },
+    variants: Object.keys((theme.vars ?? theme).palette)
+      // @ts-ignore
+      .filter((key) => (theme.vars ?? theme).palette[key].main)
+      .map((color) => ({
+        props: { color },
+        style: {
+          [`&.${pickersOutlinedInputClasses.focused}:not(.${pickersOutlinedInputClasses.error}) .${pickersOutlinedInputClasses.notchedOutline}`]:
```

Comment:
A bit "hacky", but if there is no other easy way to play around the specificity, then so be it. 🙈 👌

Reply:
My other solution was a lot worse, it was to create a variant with `props: () => true` after and but the code in it :laughing: